File: avalon-modules/avalon-dataset/avalon-dataset-model/src/main/java/io/github/lishangbu/avalon/dataset/entity/MoveCategory.java
package io.github.lishangbu.avalon.dataset.entity;

import io.github.lishangbu.avalon.hibernate.Flex;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import java.io.Serial;
import java.io.Serializable;
import lombok.Data;

/// 招式分类(MoveCategory)实体类
///
/// 表示招式的宽泛类别，用于对招式效果进行分组
///
/// @author lishangbu
/// @since 2025/08/20
@Data
@Entity
public class MoveCategory implements Serializable {
  @Serial private static final long serialVersionUID = 1L;

  /// 主键
  @Id
  @Flex
  @Column(comment = "主键")
  private Long id;

  /// 招式分类内部名称
  @Column(comment = "招式分类内部名称", length = 100)
  private String internalName;

  /// 招式分类名称
  @Column(comment = "招式分类名称", length = 100)
  private String name;

  /// 招式分类描述
  @Column(comment = "招式分类描述", length = 300)
  private String description;
}

File: modules/catalog/src/main/kotlin/io/github/lishangbu/avalon/catalog/domain/MoveCategory.kt
package io.github.lishangbu.avalon.catalog.domain


/**
 * 招式分类的领域视图。
 *
 * 这类数据对应 `catalog.move_category`，用于表达招式在更细粒度层面的效果类别，
 * 例如伤害、异常状态、回复、强制换人等，属于 Catalog 里的稳定参考数据。
 *
 * @property id 招式分类标识。
 * @property code 招式分类业务码。
 * @property name 招式分类展示名称。
 * @property description 招式分类说明，可为空。
 * @property sortingOrder 列表展示顺序。
 * @property enabled 当前招式分类是否启用。
 * @property version 乐观锁版本号。
 */
data class MoveCategory(
    val id: MoveCategoryId,
    val code: String,
    val name: String,
    val description: String?,
    val sortingOrder: Int,
    val enabled: Boolean,
    val version: Long,
)

/**
 * 招式分类的轻量摘要。
 *
 * @property id 招式分类标识。
 * @property code 招式分类业务码。
 * @property name 招式分类展示名称。
 */
data class MoveCategorySummary(
    val id: MoveCategoryId,
    val code: String,
    val name: String,
)

/**
 * 创建或更新招式分类时使用的输入草稿。
 *
 * @property code 招式分类业务码。
 * @property name 招式分类展示名称。
 * @property description 招式分类说明，可为空。
 * @property sortingOrder 列表展示顺序。
 * @property enabled 当前招式分类是否启用。
 */
data class MoveCategoryDraft(
    val code: String,
    val name: String,
    val description: String?,
    val sortingOrder: Int,
    val enabled: Boolean,
)